services/api/mw2pdf/src/classes/MediaWikiSession.ts

```diff
@@ -1,131 +1,219 @@
-    await page.goto(url, { waitUntil: 'networkidle2' })
+    );
+
+    // Get a new page
+    const page = await browser.newPage();
+
+    // Set cookies into the browser that we pull from the MW session. This means
+    // that all the login and authentication we did before will be valid.
+    await page.setCookie(...this.getCookies(domain));
+
+    // Now, go to the page!
+    await page.goto(url.toString(), { waitUntil: 'networkidle2' });
+
+    // Get the page's title.
     const pageTitle = await page.title()
-    const pagePdf = pdfGenerator
-      .generatePdfObject({ title: pageTitle })
-    await page.pdf({ path: pagePdf.path, format: 'A4' })
-    await browser.close()
-    if (includeTitlePage === true) {
-      const titlePdf = await pdfGenerator
-        .generateTitlePagePdf(pageTitle)
-      const mergedPdf = await pdfGenerator
-        .generateMergedPdf(
-          [titlePdf, pagePdf],
-          pdfGenerator.generatePdfObject({ title: pageTitle }),
-        )
-      await deletePdfs([titlePdf, pagePdf])
-      return mergedPdf
+    const pageFilename = join(options.workDirectory, `${v4()}.pdf`);
+
+    // Generate a Pdf object with the appropriate title.
+    const pagePdf: Pdf = PdfFactory.generatePdfObject(new PdfConstructorOptions(pageTitle, pageFilename));
+
+    // print the visible page into a PDF
+    await page.pdf({ path: pagePdf.filename, format: 'A4' });
+
+    // close the browser
+    await browser.close();
+
+    /**
+     * End puppeteer
+     */
+
+
+    // If the user wants a title page, let's make it!
+    if (options.makeTitlePage) {
+      const titlePagePdfFilename = join(options.workDirectory, `${v4()}.pdf`);
+      const titlePagePdf = PdfFactory.generatePdfObject(new PdfConstructorOptions(pageTitle, titlePagePdfFilename));
+      await PdfFactory.generateTitlePagePdf(pageTitle, titlePagePdf);
+
+      const titlePageAndPagePdfFilename = join(options.workDirectory, `${v4()}.pdf`);
+      const titlePageAndPagePdf = PdfFactory.generatePdfObject(new PdfConstructorOptions(pageTitle, titlePageAndPagePdfFilename));
+      await PdfFactory.generateMergedPdf([titlePagePdf, pagePdf], titlePageAndPagePdf);
+
+      await deletePdfs([titlePagePdf, pagePdf]);
+
+      return titlePageAndPagePdf;
     }
     return pagePdf
   }
 
-  async makePdfBooklet(urls, tmpDirectory = './') {
-    const pdfGenerator = new PdfGenerator(tmpDirectory)
-    // Generate temporary pdfs for each URL
-    const pagePdfs = await Promise.all(
-      urls.map(async (url) => this.makePdf(url, tmpDirectory, true)),
+  async makePdfBooklet(urls: Array<URL>, options: MwPdfOptions): Promise<Pdf> {
+
+    /**
+     * Generate a temporary PDF for each URL. Each PDF should have a
+     * title page to go with it.
+     */
+    const pagePdfs: Array<Pdf> = await Promise.all(
+      urls.map(async (url) => {
+        const pagePdfOptions = new MwPdfOptions("No Title", null, options.workDirectory, MwPdfMakeTitlePage);
```

Comment:
We've been using single quotes `'` for strings rather than double quotes; lets keep that standard (again, eslint would solve this)

Reply:
Done!